stdlib/public/runtime/SwiftObject.mm

```diff
@@ -438,7 +438,12 @@ - (BOOL)isEqual:(id)other {
     // Legacy behavior: Don't proxy to Swift Hashable or Equatable
     return NO; // We know the ids are different
   }
-
+  if (isObjCTaggedPointer(other)) {
+    // Rather than try to figure out how to deal with each possible tagged
+    // pointer type, just assume equality is symmetric and let the tagged object
+    // figure out what to do
+    return [other isEqual: self];
```

Comment:
Can this ever put us into infinite recursion?

Reply:
I've verified that all the existing tagged pointer types do not. So I think this boils down to 

* risk someone writing a new tagged pointer type that infinite recurses here
vs, for the other obvious fix of `return NO;`
* risk someone writing a new Swift class that incorrectly compares non-equal here (e.g. if someone wanted to compare equal to NSStrings)

Thoughts?

Reply:
Talked it over with Tim and I think while both of these are unlikely, the latter is much more difficult to do by accident. I'll update the patch

Reply:
I wonder if `return NO` might actually be correct here in all cases?  That would be faster.

Reply:
Pushed the `return NO` version with a comment explaining why it's correct